Description: bounds check name table "string"
 When parsing a Truetype font
 we store the name table data in a Postscript string
 and later try to extract a font name we can use for the Postscript FontName.

 We were checking for a zero length table,
 but not checking whether a non-empty string was long enough
 to contain the requested name index.

 So handle that situation gracefully.
Origin: upstream, https://git.ghostscript.com/?p=ghostpdl.git;h=d0661b8
Author: Chris Liddell <chris.liddell@artifex.com>
Bug: https://bugs.ghostscript.com/show_bug.cgi?id=703788
Last-Update: 2021-18-07
---
This patch header follows DEP-3: https://dep.debian.net/deps/dep3/
--- a/Resource/Init/gs_ttf.ps
+++ b/Resource/Init/gs_ttf.ps
@@ -244,7 +244,21 @@
   1 index length 0 gt {	% check for zero length name table
     0 1 3 index 2 getu16 1 sub {
                 % Stack: false table id index
-      12 mul 6 add 2 index exch 12 getinterval
+      12 mul 6 add 2 index exch 12
+      % Check the table is actually long enough to contain the requested index
+      1 index add 2 index length gt
+      {
+        /pdfformaterror where
+        {
+          (   **** TrueType font has malformed name table.\n) exch /pdfformaterror get exec
+          PDFSTOPONERROR {
+             /.loadpdfttfont cvx /invalidfont signalerror
+          } if
+        } if
+        pop pop
+        exit
+      } if
+      12 getinterval
       dup 6 getu16 2 index eq 1 index 8 getu16 0 ne and {
                   % We found the name we want.
         exch pop
